lib/etrade/filters.ts

```diff
@@ -0,0 +1,23 @@
+import { createEtradeGLFilter } from "@/lib/etrade/parse-etrade-gl";
+
+export const isFrQualifiedSo = createEtradeGLFilter({
+  planType: "SO",
+  qualifiedIn: "fr",
+});
+export const isUsQualifiedSo = createEtradeGLFilter({
+  planType: "SO",
+  qualifiedIn: "us",
+});
+
+export const isEspp = createEtradeGLFilter({
```

Comment:
why not distinguish fr qualified and us qualified for espp?

Reply:
ESPP is never qualified.
This is the reason acquisition gains appears on our payslip and there is a sell on acquisition day

Reply:
It can be for US-granted ESPP

Reply:
No need to add `isUsQualifiedEspp` if you don't need it but can you rename and change into `isFrQualifiedEspp`?

Reply:
From a french taxes perspective, my understanding is that it does not matter wether the plan is US qualified or not, the only thing that matters is the `planType`.

This function filters **any** ESPP related gain and losses, which makes me think the naming is correct.

Reply:
> From a french taxes perspective

Right! But when we do add support for US taxes, this'll be wrong and need updating. Anyway, not a blocker for now but I'd rather be future proof so we don't forget.

Reply:
I mean, the filter does exactly what it says: filter any ESPP G&L event.
It is independent from where it is used.
If another filter is needed, then it is totally OK to add one.

To clarify how this is used in the upcoming PR, taxes rules are applied depending on the event type, hence each rule starts by filtering the events it will be applied on.

```
const getFrTaxesForFrQualifiedSo = (events) => {
    const qualifiedFrSo = events.filter(isFrQualifiedSo);
    // ... compute related taxes
}
```

Given French tax rule for ESPP is applied on any ESPP event, US qualified or not so it starts with `events.filter(isEspp)`.

It is still possible to filter events provided to `getFrTaxesForEspp` to restrict computations to qualified espp only:

`getFrTaxesForEspp(events.filter(isUsQualifeidEspp))`

Reply:
Got it, that's helpful!